### PR TITLE
Test whether we can see ourselves rejecting invite

### PR DESCRIPTION
**Pull request by dbkr**
For a room with restricted history visibility

Regression test for https://github.com/vector-im/riot-web/issues/3462

### CODE REVIEW

File: tests/31sync/09archived.pl [hunk 1]
```diff
@@ -117,6 +117,59 @@
...
    };
 
+test "We should see our own leave event when rejecting an invite, ".
+    "even if history_visibility is restricted ",
+   requires => [ local_user_fixture( with_events => 0 ),
+                 local_user_fixture( with_events => 0 ),
+                 qw( can_sync ) ],
+
+   check => sub {
+      my ( $user1, $user2 ) = @_;
```
Comment (richvdh) [inline tests/31sync/09archived.pl]: could we call them `$invitee`, `$inviter` or something? I lost track about 15 times when reading through

File: tests/31sync/09archived.pl [hunk 2]
```diff
@@ -117,6 +117,59 @@
       });
    };
 
+test "We should see our own leave event when rejecting an invite, ".
+    "even if history_visibility is restricted ",
```
Comment (richvdh) [inline tests/31sync/09archived.pl]: spurious trailing " ". Mind you, a bug reference wouldn't hurt.

Comment (dbkr) [inline tests/31sync/09archived.pl]: done

File: tests/31sync/09archived.pl (hunk 1 above)
Comment (dbkr) [inline tests/31sync/09archived.pl]: done

**Pull request merged**